shm/Cargo.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include <string>
+
+class Cargo {
+public:
+    Cargo(const std::string& name, const size_t amount, const size_t basePrice):
+        name_(name),
+        amount_(amount),
+        basePrice_(basePrice)
+    {}
+
+    bool operator==(const Cargo& cargo) const {
+        if (name_ != cargo.getName()) {
+            return false;
+        }
+        if (amount_ != cargo.getAmount()) {
+            return false;
+        }
+        if (basePrice_ != cargo.getBasePrice()) {
+            return false;
+        }
+        return true;    
+    }
+
+    Cargo& operator+=(size_t amount) {
+        amount_ += amount;
+        return *this;
+    }
+    Cargo& operator-=(size_t amount) {
+        amount_ -= amount;
+        return *this;
+    }
```

Comment:
What happened when we reach the limit of capacity or drop below 0? You need to handle error cases. Maybe throw exception, maybe return `std::nullopt` or `nullptr` maybe set some specific value like `std::numeric_limits::min/max`.